src/Grpc.AspNetCore.Server/Internal/ClientStreamingServerCallHandler.cs

```diff
@@ -28,8 +29,15 @@ internal class ClientStreamingServerCallHandler<TRequest, TResponse, TService> :
         where TResponse : class
         where TService : class
     {
+        private delegate Task<TResponse> ClientStreamingServerCall(TService service, IAsyncStreamReader<TRequest> stream, ServerCallContext serverCallContext);
+
+        private readonly ClientStreamingServerCall _invoker;
+
         public ClientStreamingServerCallHandler(Method<TRequest, TResponse> method) : base(method)
         {
+            var handlerMethod = typeof(TService).GetMethod(Method.Name);
+
+            _invoker = (ClientStreamingServerCall)Delegate.CreateDelegate(typeof(ClientStreamingServerCall), handlerMethod);
```

Comment:
This signature should match, but if it doesn't do you think it is worth having a try/catch around creating the delegate with a friendly message.

Reply:
Yes, we should do this. We should also try to understand if it's reasonable to not match. Did they write the method by hand or is it codegen.

Reply:
It is codegen.

@jtattermusch is the method signature constant between serializers?

Reply:
@JamesNK  not sure I understand the question

Reply:
Will codegens always output the same signature for the 4 different types of calls. For example, will a unary call always be:

`Task<TResponse> SayHello(TRequest request, ServerCallContext context)`

Reply:
The server side method handlers are code generated and their signatures match these delegates:
https://github.com/grpc/grpc/blob/master/src/csharp/Grpc.Core.Api/ServerMethods.cs

https://github.com/grpc/grpc/blob/ed0a5655833e9d91a361b647db68f34e76e69774/src/csharp/Grpc.Examples/MathGrpc.cs#L284

Reply:
@JamesNK  yes, the signature on server side will always be the same and there's no plans to change that.

Reply:
Ah there are already delegates I can use!

Reply:
Ah wait, I can't use these because I need to pass in the instance (this) as the first argument. That said, maybe we don't need error handling.